src/ai/backend/client/auth.py

```diff
@@ -50,7 +51,7 @@ def generate_signature(
     Generates the API request signature from the given parameters.
     """
     hash_type = hash_type
-    hostname = endpoint._val.netloc  # type: ignore
+    hostname = urlparse(str(endpoint)).netloc
```

Comment:
I think.... we could use [`yarl.URL().authority`](https://yarl.aio-libs.org/en/latest/api/#yarl.URL.authority) as a replacement of `urlparse(...).netloc`.

```python
>>> from yarl import URL
>>> from urllib.parse import urlparse
>>> u = URL("https://user:passwd@example:9293/aihg/hohb")
>>> urlparse(str(u)).netloc
'user:passwd@example:9293'
>>> u.authority
'user:passwd@example:9293'

>>> u = URL("https://user:passwd@한글도메인.com:9293/aihg/hohb")
>>> urlparse(str(u)).netloc
'user:passwd@xn--bj0bj3i97fq8o5lq.com:9293'
>>> u.host_port_subcomponent
'xn--bj0bj3i97fq8o5lq.com:92923'
>>> u.authority
'user:passwd@한글도메인.com:9293'
```

If we upgrade yarl to 1.17 or later, we could use [`yarl.URL().host_port_subcomponent`](https://yarl.aio-libs.org/en/latest/api/#yarl.URL.host_port_subcomponent) which has the more intended semantic to use in this context.

Just for the parity with existing implementation, let's use `endpoint.authority`.

Reply:
NOTE: When we replace `netloc` with `host_port_component`, we need to check the consistency of non-ASCII hostnames with `aiohttp.web.Request().host` field used in https://github.com/lablup/backend.ai/blob/39685388/src/ai/backend/manager/api/auth.py#L372.

Reply:
We decided to use `raw_authority` here instead to maintain the behavior with the previous code.